### PR TITLE
HVG-127: Connect payin refactor

### PR DESCRIPTION
**Pull request by hedvigoscar**
- [x] Functionality is covered by an integration test
- [x] Functionality is accessible in engineering mode

Left before merging:

- [x] Replace translations with new and improved ones

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/trustly/TrustlyConnectFragment.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    private var hasLoadedWebView = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val transitionType = requireArguments().getSerializable(TRANSITION_TYPE) as? TransitionType ?: return
+
+        // TODO: Fix the directions of the animation
+        if (transitionType != TransitionType.NO_ENTER_EXIT_RIGHT) {
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/trustly/TrustlyConnectFragment.kt:40]: Should this be fixed now?

File: app/src/main/res/layout/connect_payment_result_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    tools:context=".feature.connectpayin.ConnectPaymentResultFragment"
+    android:layout_height="match_parent">
+
+    <!-- TODO: Rename these IDs to be a bit more sane -->
```
Comment (MathiasOhrstrom) [inline app/src/main/res/layout/connect_payment_result_fragment.xml:9]: Remove comment

File: app/src/main/res/layout/trustly_connect_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        android:gravity="center"
+        android:layout_weight="1">
+
+        <ProgressBar
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content" />
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:text="Loading..."
```
Comment (MathiasOhrstrom) [inline app/src/main/res/layout/trustly_connect_fragment.xml:31]: Should have translation

**Review by MathiasOhrstrom — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/trustly/TrustlyConnectFragment.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/trustly/TrustlyConnectFragment.kt:40]:
```suggestion
        if (transitionType != TransitionType.NO_ENTER_EXIT_RIGHT) {
```

File: app/src/main/res/layout/connect_payment_result_fragment.xml (hunk 2 above)
Comment (hedvigoscar) [reply app/src/main/res/layout/connect_payment_result_fragment.xml:9]:
```suggestion
```

File: app/src/main/res/layout/trustly_connect_fragment.xml (hunk 3 above)
Comment (hedvigoscar) [reply app/src/main/res/layout/trustly_connect_fragment.xml:31]: Will be added when translations are available

**Review by MathiasOhrstrom — APPROVED** [review]
Lets create add text key ticket to jira

**Comment by hedvigoscar** [comment]
I've been promised to get them exported by the end of the day, I think we can wait until then 😉

**Pull request merged**